Assets/Scripts/Augment/AugmentImplementations/RopeBody.cs

```diff
@@ -73,48 +91,150 @@ private void PullingWire()
+            {
+                if (rope.CollisionPoints.Count > 1)
+                    rope.CollisionPoints.RemoveAt(0);
+                if (rope.CollisionPoints.Count < 2)
+                { 
+                    rope.CollisionCheckActive = true;
+                    AnimateLastVertex();
+                }
+                else
+                {
+                    retractingWire.Play(audioSource);
+                }
+            })
+            .setOnCompleteOnRepeat(true)
+            .setRepeat(controlPointCount);
     }
 
     private void RemoveRope(HealthController healthController, float damage, DamageInfo info)
     {
-        isWired = false;
-        rope.enabled = false;
-        gunController.stats.Ammo = 0;
-        plugAnchor.gameObject.SetActive(false);
-        audioSource.Play();
+        if (isWired == false)
+            return;
+        RemoveRope();
     }
 
     private void CheckForWirePlanting(GunStats stats)
     {
-        if (isWired || movement.StateIsAir)
+        if (isWired || isThrowing || !canThrow)
             return;
-        plugAnchor.transform.position = gunController.Player.transform.position;
-        movement.CanMove = false;
-        movement.CanLook = false;
+
         animator.SetTrigger("Plant");
-        isWired = true;
-        rope.ResetRope(plugAnchor.WireOrigin);
+        isThrowing = true;
     }
 
     // Called by animator
     public void ActivateWire()
     {
         if (!plugAnchor)
             return;
-        rope.enabled = true;
-        plugAnchor.gameObject.SetActive(true);
-        movement.CanMove = true;
-        movement.CanLook = true;
-        gunController.stats.Ammo = gunController.stats.MagazineSize;
+        if (Physics.Raycast(gunController.Player.inputManager.transform.position, gunController.Player.inputManager.transform.forward, out RaycastHit hit, ropeLength, ignoreLayer))
+        {
+            plugAnchor.transform.position = gunController.Player.transform.position;
+            plugAnchor.transform.forward = hit.normal;
+            rope.enabled = true;
+            canThrow = false;
+            LeanTween.value(gameObject, SetThrowValue, 0f, ropeLength, 0.02f * ropeLength);
+            plugAnchor.gameObject.LeanMove(hit.point, 0.02f * hit.distance)
+                .setOnComplete(() =>
+                {
+                    rope.ResetRope(plugAnchor.WireOrigin);
+                    isThrowing = false;
+                    isWired = true;
+                    plugAnchor.SetPulseStrength(0.85f);
+                    gunController.stats.Ammo = gunController.stats.MagazineSize;
+                });
+            rope.ResetRope(plugAnchor.WireOrigin);
+            plugAnchor.gameObject.SetActive(true);
+        }
+        else
+        {
+            plugAnchor.SetPulseStrength(0);
+            plugAnchor.transform.position = gunController.Player.transform.position;
+            plugAnchor.transform.forward = gunController.Player.inputManager.transform.forward;
```

Comment:
FYI, there's a brief moment where the plug is visible on the player, I think it looks a lil weird

Reply:
Have tried hiding as best as I can now

Reply:
Seems to have worked well 👍